mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +728,461 @@ def _predict_exact_match(self, entity):
+        self.dirty = False  # never True with the current logic, kept for consistency purpose
+
+    def _predict(self, nbest_entities):
+        """Predicts the resolved value(s) for the given entity using cosine similarity.
+
+        Args:
+            nbest_entities (tuple): List of one entity object found in an input query, or a list  \
+                of n-best entity objects.
+
+        Returns:
+            (list): The top 20 resolved values for the provided entity.
+        """
+
+        syn_embs = self._cached_embs
+
+        # encode input entity
+        # TODO: Use all provided entities (i.e all nbest_entities) like elastic search
+        top_entity = nbest_entities[0]  # top_entity
+        if top_entity.text in self._cached_embs:
+            top_entity_emb = self._cached_embs[top_entity.text]
+        else:
+            top_entity_emb = self._encode(top_entity.text)[0]
+
+        try:
+            sorted_items = self._compute_cosine_similarity(syn_embs, top_entity_emb)
+            values = []
+            for synonym, score in sorted_items:
+                cnames = self._entity_mapping["synonyms"][synonym]
+                for cname in cnames:
+                    for item in self._entity_mapping["items"][cname]:
+                        item_value = copy.copy(item)
+                        item_value.pop("whitelist", None)
+                        item_value.update({"score": score})
+                        item_value.update({"top_synonym": synonym})
+                        values.append(item_value)
+        except KeyError:
+            logger.warning(
+                "Failed to resolve entity %r for type %r; "
+                "set 'clean=True' for computing embeddings of newly added items in mappings.json",
+                top_entity.text, top_entity.type
             )
-            raise EntityResolverError from e
-        except ElasticsearchException as e:
-            raise EntityResolverError from e
+            return None
+        except TypeError:
+            logger.warning(
+                "Failed to resolve entity %r for type %r", top_entity.text, top_entity.type
+            )
+            return None
+
+        return values
+
+    def _load(self):
+        self.fit()
+
+    @staticmethod
+    def _load_embeddings(cache_path):
+        """Loads embeddings for all synonyms, previously dumped into a .pkl file
+        """
+        with open(cache_path, "rb") as fp:
+            _cached_embs = pickle.load(fp)
+        return _cached_embs
+
+    @staticmethod
+    def _dump_embeddings(cache_path, data):
+        """Dumps embeddings of synonyms into a .pkl file when the .fit() method is called
+        """
+        folder = os.path.split(cache_path)[0]
+        if folder and not os.path.exists(folder):
+            os.makedirs(folder)
+        with open(cache_path, "wb") as fp:
+            pickle.dump(data, fp)
+
+
+ENTITY_RESOLVER_MODEL_MAPPINGS = {
```

Comment:
what would be the default model we use for entity resolver now?

Reply:
It is defined in `_config.py` as `DEFAULT_ENTITY_RESOLVER_CONFIG = {"model_type": "text_relevance"}`. The `.get_classifier_config()` method takes care of it at model creation time.